### PR TITLE
CEDS-1439 - preserve item list order

### PR DESCRIPTION
**Pull request by SteveSugden**

### CODE REVIEW

File: app/controllers/declaration/AdditionalFiscalReferencesController.scala [hunk 1]
```diff
@@ -111,6 +111,6 @@ class AdditionalFiscalReferencesController @Inject()(
     implicit request: JourneyRequest[AnyContent]
   ): Future[Option[ExportsDeclaration]] =
     updateExportsDeclarationSyncDirect { model =>
-      model.updatedItem(itemId, item => item.copy(additionalFiscalReferencesData = Some(updatedAdditionalFiscalReferencesData)))
+      model.updateItem(itemId, item => item.copy(additionalFiscalReferencesData = Some(updatedAdditionalFiscalReferencesData)))
```
Comment (PatrykRudnicki) [inline app/controllers/declaration/AdditionalFiscalReferencesController.scala]: What is the difference between those methods?

File: app/models/ExportsDeclaration.scala [hunk 2]
```diff
@@ -50,10 +50,9 @@ case class ExportsDeclaration(
 
   def isComplete: Boolean = status == DeclarationStatus.COMPLETE
 
-  def updatedItem(itemId: String, update: ExportItem => ExportItem): ExportsDeclaration =
-    itemBy(itemId).fold(this) { item =>
-      val updated = update(item)
-      copy(items = items.filterNot(_.id.equalsIgnoreCase(itemId)) + updated)
+  def updateItem(itemId: String, update: ExportItem => ExportItem): ExportsDeclaration =
+    itemBy(itemId).fold(this) { itemToReplace =>
+      copy(items = items.map(item => if (item.equals(itemToReplace)) update(itemToReplace) else item))
```
Comment (bambuchaAdm) [inline app/models/ExportsDeclaration.scala]: item eq itemToReplace - nice chat for reference equality

File: app/controllers/declaration/AdditionalFiscalReferencesController.scala (hunk 1 above)
Comment (SteveSugden) [reply app/controllers/declaration/AdditionalFiscalReferencesController.scala]: The letter 'd'

File: app/models/ExportsDeclaration.scala (hunk 2 above)
Comment (SteveSugden) [reply app/models/ExportsDeclaration.scala]: You're right.  Reference equality is good enough here.  I didn't even realise that scala had a third way of defining "equal" (i.e. 'eq') - I'll stick with '==' as an old Java hack.

File: app/controllers/declaration/AdditionalFiscalReferencesController.scala (hunk 1 above)
Comment (PatrykRudnicki) [reply app/controllers/declaration/AdditionalFiscalReferencesController.scala]: I thought that there are two different method for update item, didn't notice that was rename

Comment (SteveSugden) [reply app/controllers/declaration/AdditionalFiscalReferencesController.scala]: It's been renamed back.  I thought it was a typo.  Apparently its named after scala convention for collection libraries.

**Pull request merged**